driver/src/build/Driver.cpp

```diff
@@ -1245,23 +1245,24 @@ birch::Package* birch::Driver::createPackage(bool includeRequires) {
   return package;
 }
 
-void birch::Driver::readFiles(const std::string& key, bool checkExists) {
+void birch::Driver::readFiles(const std::string& key) {
   for (auto pattern : metaContents[key]) {
     auto paths = glob(pattern);
-    for (auto path : paths) {
-      if (checkExists && !exists(path)) {
-        warn(path.string() + " in build configuration does not exist.");
-      }
-      if (std::regex_search(path.string(), std::regex("\\s",
-          std::regex_constants::ECMAScript))) {
-        throw DriverException(std::string("file name ") + path.string() +
-          " in build configuration contains whitespace, which is not supported.");
-      }
-      auto inserted = allFiles.insert(path);
-      if (!inserted.second) {
-        warn(path.string() + " repeated in build configuration.");
+    if (paths.empty()) {
+        warn("no file matching '" + pattern + "' in build configuration.");
```

Comment:
As mentioned above, it does not seem useful to keep the current check when looping over `paths` since every path is guaranteed to exist. However, now the number of matching files is checked for every pattern which leads to, e.g., the warnings
```
warning: no file found matching the pattern 'src/*/*/*/*.birch' in the build configuration.
warning: no file found matching the pattern 'src/*/*/*/*/*.birch' in the build configuration.
```
when building the standard library. In general, probably users are presented with warnings when using the default patterns in the bootstrapped configuration. But maybe that's a good thing?

Reply:
Let's eliminate these warnings. Can we basically reproduce the behavior of `GLOB_NOMAGIC` by only warning when the path does not contain a wildcard `*`?

I'd like to keep all the patterns in the default `build.xml` for new packages without warnings, to make it easier for first time users. They are there as a cheap imitation of a recursive glob, i.e. `src/**.birch`, which is not supported in the standard POSIX glob.